run/jobs/index.js

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// Retrieve Job-defined env vars
+const {TASK_NUM, ATTEMPT_NUM} = process.env;
+// Retrieve User-defined env vars
+const {SLEEP_MS, FAIL_RATE} = process.env;
+
+// Define main script
+const main = async () => {
+  console.log(`Starting Task #${TASK_NUM}, Attempt #${ATTEMPT_NUM}...`);
+  // Simulate work
+  if (SLEEP_MS) {
+    await sleep(SLEEP_MS);
+  }
+  // Simulate errors
+  if (FAIL_RATE) {
+    try {
+      randomFailure(FAIL_RATE);
+    } catch (err) {
+      throw new Error(`Task #${TASK_NUM}, Attempt #${ATTEMPT_NUM} failed.`);
```

Comment:
From a pattern perspective, I like to avoid throwing a new error from inside of a catch.  This obfuscates the original callstack, and makes it very hard to figure out what actually went wrong.  Instead, try prepending your error message to the existing error, and rethrowing:

```js
err.message = `Task #${TASK_NUM}, Attempt #${ATTEMPT_NUM} failed.\n\n${err.message}`;
throw err;
```